src/lando/utils/github_checks.py

```diff
@@ -192,6 +194,277 @@ def run(
 #         raise NotImplementedError
 
 
+#
+# WARNINGS
+#
+
+
+class PullRequestWarning(PullRequestCheck, ABC):
+    """Parent class for warning checks."""
+
+
+class PullRequestBlockingReviewsWarning(PullRequestWarning):
+    """Has a review intended to block landing."""
+
+    @override
+    @classmethod
+    def run(
+        cls,
+        client: GitHubAPIClient,
+        pull_request: PullRequest,
+        target_repo: Repo,
+        request: HttpRequest,
+    ) -> list[str]:
+        reviews = pull_request.get_reviews(client)
+
+        messages = []
+
+        for review in reviews:
+            if review["state"] == pull_request.Review.CHANGES_REQUESTED:
+                messages.append(
+                    f"{cls.__doc__} {review['body'].splitlines()[0]}… {review['html_url']})"
+                )
+
+        return messages
+
+
+class PullRequestPreviouslyLandedWarning(PullRequestWarning):
+    """Has previously landed."""
+
+    @override
+    @classmethod
+    def run(
+        cls,
+        client: GitHubAPIClient,
+        pull_request: PullRequest,
+        target_repo: Repo,
+        request: HttpRequest,
+    ) -> list[str]:
+        if not pull_request.merged_at:
+            return []
+
+        return [cls.__doc__]
+
+
+class PullRequestNotAcceptedWarning(PullRequestWarning):
+    """Is not Accepted."""
+
+    @override
+    @classmethod
+    def run(
+        cls,
+        client: GitHubAPIClient,
+        pull_request: PullRequest,
+        target_repo: Repo,
+        request: HttpRequest,
+    ) -> list[str]:
+        reviews = pull_request.get_reviews(client)
+
+        if any(review["state"] == pull_request.Review.APPROVED for review in reviews):
+            return []
+
+        return [cls.__doc__]
+
+
+class PullRequestReviewsNotCurrentWarning(PullRequestWarning):
+    """No reviewer has accepted the current diff."""
+
+    @override
+    @classmethod
+    def run(
+        cls,
+        client: GitHubAPIClient,
+        pull_request: PullRequest,
+        target_repo: Repo,
+        request: HttpRequest,
+    ) -> list[str]:
+        reviews = pull_request.get_reviews(client)
+
+        if pull_request.head_sha in [
+            review["commit_id"]
+            for review in reviews
+            if review["state"] == pull_request.Review.APPROVED
+        ]:
+            return []
+
+        return [cls.__doc__]
+
+
+class PullRequestSecureRevisionWarning(PullRequestWarning):
+    """Is a secure pull request and should follow the Security Bug Approval Process."""
+
+    @override
+    @classmethod
+    def run(
+        cls,
+        client: GitHubAPIClient,
+        pull_request: PullRequest,
+        target_repo: Repo,
+        request: HttpRequest,
+    ) -> list[str]:
+        raise NotImplementedError
```

Comment:
This should be deleted, likely won't be implemented in the foreseeable future.

Reply:
```suggestion
```